asctb-api/src/functions/api.functions.ts

```diff
@@ -2,6 +2,36 @@
 import { arrayNameMap, createObject, HEADER_FIRST_COLUMN, objectFieldMap, Row } from '../models/api.model';
 import { fixOntologyId } from './lookup.functions';
 
+/**
+ *
+ * @param url  of the sheet to compare
+ * @returns a object with the sheetID, gid, and CsvUrl
+ */
+
+export function parseSheetUrl(url: string): { sheetID: string, gid: string, csvUrl: string } {
```

Comment:
Why not just return the csvurl from this function? It would make the code on the other side (and this code!) much cleaner. Maybe call it normalizeCsvUrl ?

Reply:
If the provided URL is google sheet then we need to return sheetid and gid. That's the reason I was returning sheetID, gId, and csvUrl.

Reply:
In the place where you copy/pasted it from, but not in the server where you introduced this function for the first time ;-)